web/src/app/details/DetailsPage.tsx

```diff
@@ -0,0 +1,205 @@
+            xs
+            container
+            direction='column'
+          >
+            <Grid item>
+              <CardHeader
+                title={p.title}
+                subheader={p.subheader}
+                avatar={avatar()}
+                titleTypographyProps={{
+                  'data-cy': 'title',
+                  variant: 'h5',
+                  component: 'h2',
+                }}
+                subheaderTypographyProps={{
+                  'data-cy': 'details',
+                  variant: 'body1',
+                }}
+              />
+            </Grid>
+
+            {p.details && (
+              <Grid item>
+                <CardContent className={classes.headerContent}>
+                  <Typography
+                    component='div'
+                    variant='subtitle1'
+                    color='textSecondary'
+                    data-cy='details'
+                  >
+                    <Markdown value={p.details} />
+                  </Typography>
+                </CardContent>
+              </Grid>
+            )}
+
+            <Grid className={classes.flexHeight} item />
+            <Grid item>
+              <CardActions
+                primaryActions={p.primaryActions}
+                secondaryActions={p.secondaryActions}
+              />
+            </Grid>
+          </Grid>
+        </Card>
+      </Grid>
+
+      {/* Quick Links */}
+      {p.links?.length && (
+        <Grid
+          item
+          xs={12}
+          lg={isDesktopMode(width) && p.links?.length ? 4 : 12}
+        >
+          <Card className={classes.headerCard}>
+            <CardHeader
+              title='Quick Links'
+              titleTypographyProps={{
+                variant: 'h5',
+                component: 'h2',
+              }}
+            />
+            <List data-cy='route-links' className={classes.quickLinks} dense>
+              {p.links.map((li, idx) => (
+                <ListItem
+                  key={idx}
+                  className={linkClassName(li.status)}
+                  component={AppLink}
+                  to={li.url}
+                  button
+                >
+                  <ListItemText
+                    primary={li.label}
+                    primaryTypographyProps={
+                      isDesktopMode(width) ? undefined : { variant: 'h5' }
+                    }
+                    secondary={li.subText}
+                  />
+                  <ListItemSecondaryAction>
+                    <IconButton component={AppLink} to={li.url}>
+                      <ChevronRight />
+                    </IconButton>
+                  </ListItemSecondaryAction>
+                </ListItem>
+              ))}
+            </List>
+          </Card>
+        </Grid>
+      )}
+
+      {/* Primary Page Content */}
+      {p.pageContent && (
+        <Grid
+          className={
+            isWidthDown('sm', width) ? classes.smPageBottom : undefined
```

Comment:
I'm not sure how classnames handles incoming class strings, so I used undefined to skip any processing since we already know we don't want to apply any styles